### PR TITLE
Add new spark patch for v1.6.1 with dyn-allocation support

### PR DESCRIPTION
**Pull request by solnic**
refs twosigma/cook#111

### CODE REVIEW

**Comment by leifwalsh** [comment]
LGTM but this should remove the old patch probably

**Comment by solnic** [comment]
Ah ok I just thought that maybe for some people who still run older spark this might be useful

**Comment by leifwalsh** [comment]
This is why god gave us version control :wink:

**Comment by wyegelwel** [comment]
solnic, is this ok to merge or is there more commits coming?

**Comment by solnic** [comment]
@wyegelwel I'd love to hear from you re issues I described in basecamp before merging this in

File: spark/0001-Add-cook-support-for-spark-v1.6.1.patch [hunk 1]
```diff
@@ -0,0 +1,769 @@
...
++  // want to get into thread management issues)
++  override def start() {
++    super.start()
++    logInfo("Starting Cook Spark Scheduler")
++    requestRemainingCores()
++  }
++
++  override def stop() {
++    super.stop()
++    abortJobs(runningJobUUIDs.toList)
```
Comment (solnic) [inline spark/0001-Add-cook-support-for-spark-v1.6.1.patch]: We *probably* should not be aborting jobs here, as it results in killing all jobs that the cook scheduler thinks are still running. I've tested it w/o this line and the behavior seems to be correct.

Comment (wyegelwel) [inline spark/0001-Add-cook-support-for-spark-v1.6.1.patch]: Where correct means the jobs complete successfully

Comment (wyegelwel) [inline spark/0001-Add-cook-support-for-spark-v1.6.1.patch]: Spoke with Leif. I think it is ok to just remove this abort. Once that is done, I think this PR is good to go.

**Comment by solnic** [comment]
@wyegelwel I updated the patch so that it longer includes the aborting of jobs

**Comment by wyegelwel** [comment]
LGTM, @leifwalsh @WenboZhao unless you have any other comments I think this is good to merge.

**Comment by leifwalsh** [comment]
:100:

**Pull request merged**